delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/Archives.scala

```diff
@@ -1,24 +1,195 @@
+      iri <- expandIri(id, p)
+      res <- eval(CreateArchive(iri, project, value, subject), p)
+    } yield res).named("createArchive", moduleType)
+
+  /**
+    * Creates an archive from a json-ld representation. If an id is detected in the source document it will be used.
+    * Alternately, an id is generated by the system.
+    *
+    * @param project the archive parent project
+    * @param source  the archive json representation
+    * @param subject the subject that initiated the action
+    */
+  def create(project: ProjectRef, source: Json)(implicit subject: Subject): IO[ArchiveRejection, ArchiveResource] =
+    (for {
+      p            <- projects.fetchActiveProject(project)
+      (iri, value) <- sourceDecoder(p, source)
+      res          <- eval(CreateArchive(iri, project, value, subject), p)
+    } yield res).named("createArchive", moduleType)
+
+  /**
+    * Creates an archive from a json-ld representation with a user specified id. If an id is also detected in the source
+    * document it will be compared with the specified id. If the user specified id does not match the detected source
+    * id, the call will be rejected.
+    *
+    * @param id      the archive identifier
+    * @param project the archive parent project
+    * @param source  the archive json representation
+    * @param subject the subject that initiated the action
+    */
+  def create(
+      id: IdSegment,
+      project: ProjectRef,
+      source: Json
+  )(implicit subject: Subject): IO[ArchiveRejection, ArchiveResource] =
+    (for {
+      p     <- projects.fetchActiveProject(project)
+      iri   <- expandIri(id, p)
+      value <- sourceDecoder(p, iri, source)
+      res   <- eval(CreateArchive(iri, project, value, subject), p)
+    } yield res).named("createArchive", moduleType)
+
+  /**
+    * Fetches an existing archive.
+    *
+    * @param id      the archive identifier
+    * @param project the archive parent project
+    */
+  def fetch(id: IdSegment, project: ProjectRef): IO[ArchiveRejection, ArchiveResource] =
+    (for {
+      p     <- projects.fetchProject(project)
+      iri   <- expandIri(id, p)
+      state <- currentState(project, iri)
+      res   <- IO.fromOption(state.toResource(p.apiMappings, p.base, cfg.ttl), ArchiveNotFound(iri, project))
+    } yield res).named("fetchArchive", moduleType)
+
+  // TODO: implement download function
+  //def download(id: IdSegment, projectRef: ProjectRef): IO[ArchiveRejection, Source[ByteString, Any]] = ???
+
+  private def eval(cmd: CreateArchive, project: Project): IO[ArchiveRejection, ArchiveResource] =
+    (for {
+      result    <- aggregate.evaluate(identifier(cmd.project, cmd.id), cmd).mapError(_.value)
+      (am, base) = project.apiMappings -> project.base
+      resource  <- IO.fromOption(result.state.toResource(am, base, cfg.ttl), UnexpectedInitialState(cmd.id, project.ref))
+    } yield resource).named("evaluateArchiveCommand", moduleType)
```

Comment:
this span is interesting, but it is not present in any other module. We should be consistent so either remove it add it everywhere else

Reply:
the evaluate command in sourcing already has it, so I'll remove it